tests/crypto_test.py

```diff
@@ -26,3 +56,286 @@ def test_returned_string_is_lower_case(self):
+        ret = crypto.check_uid_validity(key, mock.sentinel.EMAIL)
+        self.assertFalse(ret)
+
+
+class TestListKeys(_TestKey):
+
+    def test_list_no_hints(self):
+        # This only tests that you get 3 keys back (the number in our test
+        # keyring), it might be worth adding tests to check more about the keys
+        # returned
+        values = crypto.list_keys()
+        self.assertEqual(len(list(values)), 3)
+
+    def test_list_hint(self):
+        values = crypto.list_keys(hint="ambig")
+        self.assertEqual(len(list(values)), 2)
+
+
+class TestGetKey(_TestKey):
+
+    def test_plain(self):
+        # Test the uid of the only identity attached to the key we generated.
+        expected = self.ctx.get_key(self.FPR).uids[0].uid
+        actual = crypto.get_key(self.FPR).uids[0].uid
+        self.assertEqual(expected, actual)
+
+    def test_validate(self):
+        # Since we already test validation we're only going to test validate
+        # once.
+        expected = self.ctx.get_key(self.FPR).uids[0].uid
+        actual = crypto.get_key(self.FPR, validate=True, encrypt=True, sign=True).uids[0].uid
+        self.assertEqual(expected, actual)
+
+    def test_invalid_key(self):
+        with self.assertRaises(GPGProblem) as caught:
+            crypto.get_key('z')
+        self.assertEqual(caught.exception.code, GPGCode.NOT_FOUND)
+
+    @mock.patch('alot.crypto.check_uid_validity', mock.Mock(return_value=True))
+    def test_signed_only_true(self):
+        try:
+            crypto.get_key(self.FPR, signed_only=True)
+        except GPGProblem as e:
+            raise AssertionError(e)
+
+    @mock.patch('alot.crypto.check_uid_validity', mock.Mock(return_value=False))
+    def test_signed_only_false(self):
+        with self.assertRaises(GPGProblem) as e:
+            crypto.get_key(self.FPR, signed_only=True)
+        self.assertEqual(e.exception.code, GPGCode.NOT_FOUND)
+
+    @staticmethod
+    def _context_mock():
+        error = gpgme.GpgmeError()
+        error.code = gpgme.ERR_AMBIGUOUS_NAME
+        context_mock = mock.Mock()
+        context_mock.get_key = mock.Mock(side_effect=error)
+
+        return context_mock
+
+    def test_ambiguous_one_valid(self):
+        invalid_key = make_key(invalid=True)
+        valid_key = make_key()
+
+        with mock.patch('alot.crypto.gpgme.Context',
+                        mock.Mock(return_value=self._context_mock())), \
+                mock.patch('alot.crypto.list_keys',
+                           mock.Mock(return_value=[valid_key, invalid_key])):
+            key = crypto.get_key('placeholder')
+        self.assertIs(key, valid_key)
+
+    def test_ambiguous_two_valid(self):
+        with mock.patch('alot.crypto.gpgme.Context',
+                        mock.Mock(return_value=self._context_mock())), \
+                mock.patch('alot.crypto.list_keys',
+                           mock.Mock(return_value=[make_key() for _ in xrange(2)])):
```

Comment:
Eh, it's either way. I can change it if you think `[make_key(), make_key()]` is more readable.